vir/defs/polymorphic/stmt.rs

```diff
@@ -0,0 +1,78 @@
+// © 2019, ETH Zurich
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use super::*;
+use prusti_common::vir::borrows::{Borrow, DAG as ReborrowingDAG};
```

Comment:
Please also copy `borrows` so that we can remove dependency on `prusti_common`.

Reply:
I see. Where would be a good place to put `borrows`? In [`defs`](https://github.com/Maxwell-Yang-2001/prusti-dev/tree/polymorphic-vir/vir/defs) (Same level as the `polymorphic` directory)?

Reply:
Let's put it inside `vir/src`. Then, the following import should work:

```rust
use crate::borrows::{Borrow, DAG as ReborrowingDAG};
```

Reply:
And this `borrows.rs` also relies on part of `polymorphic`. By being placed outside of `polymorphic` (so without being generated as part of `vir.rs`), how would it import things from `vir.rs`? I think the main confusion for me is how `vir.rs` would be located relative to other files or directories in the `vir` crate.

Reply:
Oh, I missed that `borrows.rs` depends on VIR. In that case, it needs to be inside `defs/polymorphic`.

Reply:
> I think the main confusion for me is how vir.rs would be located relative to other files or directories in the vir crate.

The generated `vir.rs` is put into the `target` directory. It is included by the include macro in `vir/src/lib.rs`, so you can think as all code from `vir.rs` is inlined at that place.